File: Chapter1/Binary_search.cpp
#include<iostream>
using namespace std;
int binarySearch(int arr[], int p, int r, int number){
	if(p<=r){
		int mid=(p+r)/2;
		if (arr[mid] == number)
		return mid;
		if(arr[mid]>number)
		return binarySearch(arr,p,mid-1,number);
	    if(arr[mid]<number)
		return binarySearch(arr,mid+1,r,number);
		
	}
	return -1;
}
int main(void){
	int arr[]={1,3,7,15,18,20,25,30,36,40};
	int n= sizeof(arr)/sizeof(arr[0]);
	int number;
	cout<<"enter the number to search:\n";
	cin >> number;
	int index=binarySearch(arr,0,n-1,number);
	if(index == -1){
		cout<<number<<"is not present in the array";
	}
	else{
		cout << number << "is present at index" << index << "in the array";
	}
	return 0;
}

File: Chapter1/Binary_search.py
import math
def BinarySearch(arr1, m, key):
    print(f"sorted array is : {arr1}")
    start = 0
    end = m-1
    while start <= end:
        mid = math.floor(start+end)//2
        if arr1[mid] == key:
            return 1
        elif arr1[mid] < key:
            start = mid+1
        elif arr1[mid] > key:
            end = mid-1
    return 0

m = int(input("Enter the size of array :"))

arr = []
print(f"Enter the {n} element in array")
for i in range(0, m):
    ele = int(input())
    arr.append(ele)

key = int(input("Enter the searching element "))
arr1 = sorted(arr)
print(f"sorted array is : {arr1}")
if(BinarySearch(arr1, m, key)==1):
    print("Element found")
else:
    print("Element not found")
